contrib/src/main/scala/akka/stream/contrib/SourceGen.scala

```diff
@@ -7,27 +7,35 @@ import akka.stream._
 import akka.stream.scaladsl._
 import akka.stream.stage._
 import com.typesafe.config.ConfigFactory
-import scala.concurrent.duration.Duration
+import scala.concurrent.duration.{ Duration, FiniteDuration }
 
 /**
  * Source factory methods are placed here
  */
 object SourceGen {
+  private[this] lazy val defaultTimeout = Duration.fromNanos(ConfigFactory.load().getDuration("akka.stream.contrib.unfold-flow-timeout").toNanos)
```

Comment:
Loading the config separately is not a good idea, it should come from the actor system. I see two reasonable options:
1. Have the stage/logic take an option, and if that is empty get the system.settings.config from the running stream
2. Just skip the config part entirely (because there is no reasonable way to have a default as it depends on the application)